ruby/ql/lib/codeql/ruby/frameworks/rack/internal/App.qll

```diff
@@ -2,24 +2,44 @@
  * Provides modeling for Rack applications.
  */
 
+private import codeql.ruby.AST
 private import codeql.ruby.ApiGraphs
 private import codeql.ruby.DataFlow
 private import codeql.ruby.typetracking.TypeTracker
 private import Response::Private as RP
 
-/** A method node for a method named `call`. */
-private class CallMethodNode extends DataFlow::MethodNode {
-  CallMethodNode() { this.getMethodName() = "call" }
+/**
+ * A callable node that takes a single argument and, if it has a method name,
+ * is called "call".
+ */
+private class PotentialRequestHandler extends DataFlow::CallableNode {
+  PotentialRequestHandler() {
+    this.getNumberOfParameters() = 1 and
+    (
+      this.(DataFlow::MethodNode).getMethodName() = "call"
+      or
+      not this instanceof DataFlow::MethodNode and
+      exists(DataFlow::CallNode cn | cn.getMethodName() = "run" |
+        this.(DataFlow::LocalSourceNode).flowsTo(cn.getArgument(0))
+        or
+        // TODO: `Proc.new` should automatically propagate flow from its block argument
```

Comment:
I've opened a PR here to fix this https://github.com/github/codeql/pull/13612

After that gets merged we should be able to write the entire following clause as:
```codeql
this = API::getTopLevelCall("run").getArgument(0).asCallable()
```

Reply:
I'll drop the explicit proc handling once that PR is in.

Reply:
Since the PR is not able to be merged right now, let's leave it like this for now